internal/monitors/statsd/statsd.go

```diff
@@ -0,0 +1,160 @@
+package statsd
+
+import (
+	"net"
+	"strconv"
+	"strings"
+)
+
+type statsDListener struct {
+	ipAddr       string
+	port         uint16
+	tcp          bool
+	udpConn      *net.UDPConn
+	tcpConn      *net.TCPConn
+	prefix       string
+	converters   []*converter
+	metricBuffer []string
+}
+
+type statsDMetric struct {
+	metricName string
+	metricType string
+	value      float64
+	dimensions map[string]string
+}
+
+func (sl *statsDListener) Listen() error {
+	if sl.tcp {
+		return sl.listenTCP()
+	}
+
+	return sl.listenUDP()
+}
+
+func (sl *statsDListener) listenUDP() error {
+	conn, err := net.ListenUDP("udp", &net.UDPAddr{
+		IP:   net.ParseIP(sl.ipAddr),
+		Port: int(sl.port),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	sl.udpConn = conn
+	return nil
+}
+
+func (sl *statsDListener) listenTCP() error {
+	return nil
+}
+
+func (sl *statsDListener) FetchMetrics() []*statsDMetric {
+	rawMetrics := make([]string, len(sl.metricBuffer))
+
+	copy(rawMetrics, sl.metricBuffer)
+	sl.metricBuffer = nil
+
+	parsed := parseMetrics(rawMetrics, sl.converters, sl.prefix)
+
+	return parsed
+}
+
+func (sl *statsDListener) Read() {
+	chData := make(chan []byte)
+
+	if sl.tcp {
+		go sl.readTCP(chData)
+	} else {
+		go sl.readUDP(chData)
+	}
+
+	for {
+		select {
+		case data := <-chData:
+			sl.metricBuffer = append(sl.metricBuffer, strings.Split(string(data), "\n")...)
+		}
+	}
+}
+
+func (sl *statsDListener) readTCP(chData chan []byte) {
+}
+
+func (sl *statsDListener) readUDP(chData chan []byte) {
+	// UDP needs to receive data packet by packet. Max packet size is 65535 for now.
```

Comment:
IPv4 UDP max packet size is a bit under 64k but for IPv6 it can be much bigger.

Reply:
@keitwb Did you mean this? https://en.wikipedia.org/wiki/Jumbogram

Reply:
Looks like pretty much standard https://github.com/prometheus/statsd_exporter/commit/b8aab013c6538118225b031a46dafc344290debe#diff-557eb2a359922e8de5f18397fed0cd99